SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -58,2355 +59,2373 @@ using namespace std;
-  su2double AuxVar;      /*!< \brief Auxiliar variable for gradient computation. */
-  su2double *Grad_AuxVar;  /*!< \brief Gradient of the auxiliar variable. */
-  su2double Delta_Time;  /*!< \brief Time step. */
-  su2double Max_Lambda,  /*!< \brief Maximun eingenvalue. */
-  Max_Lambda_Inv,    /*!< \brief Maximun inviscid eingenvalue. */
-  Max_Lambda_Visc,  /*!< \brief Maximun viscous eingenvalue. */
-  Lambda;        /*!< \brief Value of the eingenvalue. */
-  su2double Sensor;  /*!< \brief Pressure sensor for high order central scheme and Roe dissipation. */
-  su2double *Undivided_Laplacian;  /*!< \brief Undivided laplacian of the solution. */
-  su2double *Res_TruncError,  /*!< \brief Truncation error for multigrid cycle. */
-  *Residual_Old,    /*!< \brief Auxiliar structure for residual smoothing. */
-  *Residual_Sum;    /*!< \brief Auxiliar structure for residual smoothing. */
-  static unsigned short nDim;    /*!< \brief Number of dimension of the problem. */
-  unsigned short nVar;    /*!< \brief Number of variables of the problem,
-                           note that this variable cannnot be static, it is possible to
-                           have different number of nVar in the same problem. */
-  unsigned short nPrimVar, nPrimVarGrad;    /*!< \brief Number of variables of the problem,
-                                             note that this variable cannnot be static, it is possible to
-                                             have different number of nVar in the same problem. */
-  unsigned short nSecondaryVar, nSecondaryVarGrad;    /*!< \brief Number of variables of the problem,
-                                                       note that this variable cannnot be static, it is possible to
-                                                       have different number of nVar in the same problem. */
-  su2double *Solution_Adj_Old;    /*!< \brief Solution of the problem in the previous AD-BGS iteration. */
-  
-  int *Input_AdjIndices,          /*!< \brief Indices of Solution variables in the adjoint vector. */
-  *Output_AdjIndices;             /*!< \brief Indices of Solution variables in the adjoint vector after having been updated. */
-  
-  /*--- Old solution container for BGS iterations ---*/
+    su2double& operator() (unsigned long i, unsigned long j, unsigned long k) { return storage(i*M*N + j*N + k); }
+    const su2double& operator() (unsigned long i, unsigned long j, unsigned long k) const { return storage(i*M*N + j*N + k); }
+
+    su2double** operator[] (unsigned long i) { return interface[i]; }
+  };
+
+  MatrixType Solution;       /*!< \brief Solution of the problem. */
+  MatrixType Solution_Old;   /*!< \brief Old solution of the problem R-K. */
+
+  MatrixType External;       /*!< \brief External (outer) contribution in discrete adjoint multizone problems. */
+  MatrixType External_Old;   /*!< \brief Old external (outer) contribution in discrete adjoint multizone problems. */
```

Comment:
This whole section looks very tidy and clean. Nice

Reply:
Thanks, sadly I broke the cht adjoint... residuals during recording are the same but I get NaN's on the first iteration, I may have to redo everything as I have no idea what to look for. Any clues?

Reply:
I'll have a look right now

Reply:
So I ran a single zone case (fluid) with the multizone discrete adjoint driver and all looks good. Which tells us that it's unlikely that something is wrong with your adaptions of e.g. `CVariable::SetAdjIndices`.
Just removed the routine that throws that error to see that extracting adjoints in the heat zones seem to fail. I'll keep testing different settings..

Reply:
Thank you for chasing this! I pushed that fix, if it passes I think we should merge as travis is currently failing in develop (#801) because I updated one testcase.
I fixed 3 uninitialized warnings, maybe those are the ones you are referring to?